grakn-core/src/main/java/ai/grakn/util/CommonUtil.java

```diff
@@ -160,4 +162,47 @@ public static RuntimeException unreachableStatement(@Nullable String message, Th
             }
         };
     }
+
+    public static <T, K, V> Collector<T, ?, HashMultimap<K, V>> flatteningToMultimap(
+            Function<? super T, ? extends K> keyFunction,
+            Function<? super T, ? extends Stream<? extends V>> valueFunction) {
+        return flatteningToMultimap(keyFunction, valueFunction, HashMultimap::create);
+    }
+
+    public static <T, K, V, M extends Multimap<K, V>> Collector<T, ?, M> flatteningToMultimap(
```

Comment:
What???? All these generics make my eyes bleed. I have been staring at this for 10 mins and I am still not sure I fully get it. The end result is clear but could you maybe make the process a bit clearer?

Some ways you could clarify this:
1. Define the purpose of each function
2. Provide an example usage in the comments